src/lib/components/dashboard/parserHelper.ts

```diff
@@ -0,0 +1,354 @@
+		.filter((credential) => credential.type === "login")
+		.map((credential) => ({
+			name: credential.name,
+			description: credential.notes,
+			domain: credential.login_uri,
+			username: credential.login_username,
+			password: credential.login_password,
+			totp: credential.login_totp,
+		}));
+}
+
+export function transformProtonpassCredentials(parsedData: Credential[]) {
+	return parsedData
+		.filter(isProtonCredential)
+		.filter((credential) => credential.type === "login")
+		.map((credential) => {
+			const extractedUsername = extractUsername(
+				credential.username,
+				credential.email,
+			);
+
+			const result: IntermediateCredential = {
+				name: credential.name,
+				description: credential.note,
+				domain: credential.url,
+				username: extractedUsername,
+				password: credential.password,
+			};
+
+			if (extractedUsername === credential.username && credential.email) {
+				result.email = credential.email;
+			}
+
+			if (credential.totp) {
+				const parsedTotp = extractTOTPSecret(credential.totp);
+				if (parsedTotp) {
+					result.totp = parsedTotp;
+				}
+			}
+
+			return result;
+		});
+}
+
+export function transformDashlaneCredentials(parsedData: Credential[]) {
+	return parsedData.filter(isDashlaneCredential).map((credential) => ({
+		name: credential.title,
+		description: credential.note,
+		domain: credential.url,
+		username: credential.username,
+		password: credential.password,
+		totp: credential.otpSecret,
+	}));
+}
+
+export function transformNordpassCredentials(parsedData: Credential[]) {
+	return parsedData
+		.filter(isNordpassCredential)
+		.filter((credential) => credential.type === "password")
+		.map((credential) => ({
+			name: credential.name,
+			description: credential.note,
+			domain: credential.url,
+			username: credential.username,
+			password: credential.password,
+		}));
+}
+
+export function transformKeepassCredentials(parsedData: Credential[]) {
+	return parsedData.filter(isKeepassCredential).map((credential) => ({
+		name: `Login - ${new URL(credential["Web Site"]).hostname}`,
+		description: credential.Comments,
+		domain: credential["Web Site"],
+		username: credential["Login Name"],
+		password: credential.Password,
+	}));
+}
+
+export function transformRoboformCredentials(parsedData: Credential[]) {
+	return parsedData.filter(isRoboformCredential).map((credential) => ({
+		name: credential.Name,
+		description: credential.Note,
+		domain: credential.Url,
+		username: credential.Login,
+		password: credential.Pwd,
+	}));
+}
+
+export function transformOnepasswordCredentials(parsedData: Credential[]) {
+	return parsedData.filter(is1passwordCredential).map((credential) => ({
+		name: credential.Title,
+		description: credential.Notes,
+		domain: credential.Url,
+		username: credential.Username,
+		password: credential.Password,
+		totp: credential.OTPAuth,
+	}));
+}
+
+export async function finalProcessing(
+	folderId: string,
+	usersToShare: UsersForDataSync[],
+	{
```

Comment:
use a type here and dont destructure, use obj. notation, helps keeping the arguments clean